runtime/oti/VMHelpers.hpp

```diff
@@ -2114,19 +2124,31 @@ class VM_VMHelpers
  *   another GC thread winning to scan(bit0/bit1,bit0 and bit1 is irrelevant and independent), again don't do anything, and let the winning thread do the work, instead
  */
 	static VMINLINE bool
-	tryWinningConcurrentGCScan(J9VMContinuation *continuation, bool isGlobalGC)
+	tryWinningConcurrentGCScan(J9VMContinuation *continuation, bool isGlobalGC, bool beingMounted)
 	{
-		uintptr_t complementGCConcurrentState = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
-		uintptr_t returnedState = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
+		volatile uintptr_t *localAddr = &continuation->state;
 		do {
-			/* preserve the concurrent GC state for the other type of GC */
-			complementGCConcurrentState = continuation->state & getConcurrentGCMask(!isGlobalGC);
-			returnedState = VM_AtomicSupport::lockCompareExchange(&continuation->state, complementGCConcurrentState, complementGCConcurrentState | getConcurrentGCMask(isGlobalGC));
-		/* if the other GC happened to change its concurrentGC state since us taking a snapshot of their state, we'll have to retry */
-		} while (complementGCConcurrentState != (returnedState & complementGCConcurrentState));
+			uintptr_t continuationState = *localAddr;
 
-		/* if returned state does not contain carrier ID, return that we won */
-		return (complementGCConcurrentState == returnedState);
+			/* If it's being concurrently scanned within the same type of GC by another thread , it's unnecessary to do it again */
+			if (!isConcurrentlyScannedFromContinuationState(continuationState, isGlobalGC)) {
+				/* If it's fully mounted, it's unnecessary to scan now, since it will be compensated by pre/post mount actions.
+					If it's being mounted by this thread, we must be in pre/post mount (would be nice, but not trivial to assert it),
+					therefore we must scan.
+				 */
+				if (beingMounted || !isContinuationMountedFromContinuationState(continuationState)) {
+					/* Try to set scan bit for this GC type */
+					uintptr_t returnedState = VM_AtomicSupport::lockCompareExchange(localAddr, continuationState, continuationState | getConcurrentGCMask(isGlobalGC));
+					/* If no other thread changed anything (mounted or won scanning for any GC), we succeeded, otherwise retry */
+					if (continuationState == returnedState) {
+						return true;
+					} else {
+						continue;
+					}
+				}
+			}
+		} while (false);
+		return false;
```

Comment:
add a comment like this: 

We did not even try to win, since it was either mounted or already being scanned